pinot-common/src/main/java/org/apache/pinot/common/request/context/LiteralContext.java

```diff
@@ -162,7 +162,17 @@ public Object getValue() {
   public boolean getBooleanValue() {
     Boolean booleanValue = _booleanValue;
     if (booleanValue == null) {
-      booleanValue = _pinotDataType != null && _pinotDataType.toBoolean(_value);
+      if (_value instanceof Boolean) {
```

Comment:
Why do we need this? Are you trying to get boolean out of `NULL`? This won't be accurate because `NULL` is not equivalent to boolean `false`

Reply:
This issue is that boolean Literal is converted to 1 or 0 value
Ref: https://github.com/apache/pinot/blob/master/pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RexExpressionUtils.java#L70

Reply:
for null, it anyway won't work with type `boolean`.

Reply:
Changed to only handle the Integer conversion.

Reply:
The bug is on the `RexExpression` conversion side, where boolean value is not properly preserved. We should change `RexExpressionUtils` to keep the value type aligned with the data type. Also, if I read it correctly, `TIMESTAMP` is also not properly handled as it is not supported within `LiteralContext` yet